### PR TITLE
Rename directories Linux and macOS

### PR DESCRIPTION
**Pull request by DFolchA**
|Related issue|
|---|
|#570|

## Description
This PR adds the necessary changes to rename the  `logs/ossec` and `queue/ossec` directories to DEB, RPM, AIX, and macOS packages.

## Tests

- Build the package in any supported platform
  - [x] Linux
  - [x] macOS
  - [x] AIX
- [x] Package installation
- [x] Package upgrade
- [x] Package downgrade
- [x] Package remove
- [x] Package install/remove/install
- [ ] Change added to CHANGELOG.md

- Tests for Linux RPM
  - [x] Build the package for x86_64
  - [x] Build the package for i386
  - [x] Build the package for armhf
  - [x] Build the package for aarch64
  - [x] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [x] Build the package for x86_64
  - [x] Build the package for i386
  - [x] Build the package for armhf
  - [x] Build the package for aarch64
  - [x] Package install/remove/install
  - [x] Package install/purge/install
  - [x] Check file permissions after installing the package
- Tests for macOS
  - [ ] Test the package from macOS Sierra to Mojave
- Tests for IBM AIX
  - [x] `%files` section is correctly updated if necessary
  - [ ] Check the changes from IBM AIX 5 to 7

### CODE REVIEW

File: rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec [hunk 1]
```diff
@@ -526,6 +535,14 @@ if [ -f %{_localstatedir}/tmp/wazuh.restart ]; then
...
 fi
 
+if [ -d %{_localstatedir}/logs/ossec ]; then
+  rm -rf %{_localstatedir}/logs/ossec/
+fi
+
+if [ -d %{_localstatedir}/queue/ossec ]; then
+  rm -rf %{_localstatedir}/queue/ossec/
+fi
+
```
Comment (okynos) [inline rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec:545]: Update also the files section like 4.2.0 SPEC

File: aix/SPECS/4.2.0/wazuh-agent-4.2.0-aix.spec [hunk 2]
```diff
@@ -101,11 +101,23 @@ if [ $1 = 1 ]; then
 fi
 
 if [ $1 = 2 ]; then
-  touch %{_localstatedir}/tmp/wazuh.restart
-  /etc/rc.d/init.d/wazuh-agent stop > /dev/null 2>&1 || :
+  if %{_localstatedir}/bin/wazuh-control status 2>/dev/null | grep "is running" > /dev/null 2>&1; then
```
Comment (okynos) [inline aix/SPECS/4.2.0/wazuh-agent-4.2.0-aix.spec:104]: What happens if `wazuh-control` doesn't exists?

File: aix/SPECS/5.0.0/wazuh-agent-5.0.0-aix.spec [hunk 3]
```diff
@@ -196,6 +208,14 @@ if [ -f %{_localstatedir}/tmp/wazuh.restart ]; then
...
 fi
 
+if [ -d %{_localstatedir}/logs/ossec ]; then
+  rm -rf %{_localstatedir}/logs/ossec/
+fi
+
+if [ -d %{_localstatedir}/queue/ossec ]; then
+  rm -rf %{_localstatedir}/queue/ossec/
+fi
+
```
Comment (okynos) [inline aix/SPECS/5.0.0/wazuh-agent-5.0.0-aix.spec:218]: The files section needs to be updated.

**Review by okynos — APPROVED** [review]
Minor changes :D

File: rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec (hunk 1 above)
Comment (DFolchA) [reply rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec:545]: That issue was fixed in this PR https://github.com/wazuh/wazuh-packages/pull/643

File: aix/SPECS/4.2.0/wazuh-agent-4.2.0-aix.spec (hunk 2 above)
Comment (DFolchA) [reply aix/SPECS/4.2.0/wazuh-agent-4.2.0-aix.spec:104]: That issue was fixed in this PR #643

File: aix/SPECS/5.0.0/wazuh-agent-5.0.0-aix.spec (hunk 3 above)
Comment (DFolchA) [reply aix/SPECS/5.0.0/wazuh-agent-5.0.0-aix.spec:218]: Done in 70519fd

File: aix/SPECS/4.2.0/wazuh-agent-4.2.0-aix.spec [hunk 4]
```diff
@@ -101,11 +101,23 @@ if [ $1 = 1 ]; then
...
+    /etc/rc.d/init.d/wazuh-agent stop > /dev/null 2>&1 || :
+    touch %{_localstatedir}/tmp/wazuh.restart
+  fi
 fi
 
 %post
+if [ $1 = 2 ]; then
+  if [ -d %{_localstatedir}/logs/ossec ]; then
+    cp -rT %{_localstatedir}/logs/ossec/ %{_localstatedir}/logs/wazuh
```
Comment (alberpilot) [inline aix/SPECS/4.2.0/wazuh-agent-4.2.0-aix.spec:113]: This needs to be tested. https://www.ibm.com/support/knowledgecenter/ssw_aix_72/c_commands/cp.html
Flag `T` doesn't appear in the AIX cp command guide. Request AIX 6 if needed.

File: rpms/SPECS/4.2.0/wazuh-agent-4.2.0.spec [hunk 5]
```diff
@@ -435,6 +443,14 @@ if [ -f %{_localstatedir}/tmp/wazuh.restart ]; then
   fi
 fi
 
+if [ -d %{_localstatedir}/logs/ossec ]; then
```
Comment (alberpilot) [inline rpms/SPECS/4.2.0/wazuh-agent-4.2.0.spec:446]: What if we include an `and` in the if condition about `%{_localstatedir}/logs/wazuh` exist? For `%{_localstatedir}/logs/ossec/` removal the condition must be both folders exist, right? Just to be sure.

File: rpms/SPECS/4.2.0/wazuh-agent-4.2.0.spec [hunk 6]
```diff
@@ -435,6 +443,14 @@ if [ -f %{_localstatedir}/tmp/wazuh.restart ]; then
...
+  rm -rf %{_localstatedir}/logs/ossec/
+fi
+
+if [ -d %{_localstatedir}/queue/ossec ]; then
```
Comment (alberpilot) [inline rpms/SPECS/4.2.0/wazuh-agent-4.2.0.spec:450]: Same

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes

File: aix/SPECS/4.2.0/wazuh-agent-4.2.0-aix.spec (hunk 4 above)
Comment (DFolchA) [reply aix/SPECS/4.2.0/wazuh-agent-4.2.0-aix.spec]: Done in 38e5863

File: rpms/SPECS/4.2.0/wazuh-agent-4.2.0.spec (hunk 5 above)
Comment (DFolchA) [reply rpms/SPECS/4.2.0/wazuh-agent-4.2.0.spec:446]: These changes only apply from 4.2.0 onward so the new directories will always be present.

File: rpms/SPECS/4.2.0/wazuh-agent-4.2.0.spec (hunk 6 above)
Comment (DFolchA) [reply rpms/SPECS/4.2.0/wazuh-agent-4.2.0.spec:450]: Same

**Comment by DFolchA** [comment]
Test using real rotated logs:

##RPM
image

##DEB
image

**Review by okynos — APPROVED** [review]
LGTM.

**Pull request merged**